src/ranged.cpp

```diff
@@ -166,14 +166,41 @@ bool player::handle_gun_damage( item &it, int shots_fired )
 
     const auto &curammo_effects = it.ammo_effects();
     const cata::optional<islot_gun> &firing = it.type->gun;
+    // Here we check for a chance for the weapon to suffer a misfire due to
+    // being dirty (blackpowder fouling effect only; typical dirt accumulation
+    // from smokeless powder is not currently modeled by this effect).
+    // Can damage gun, but not enough to push the weapon beyond 'shattered'.
+    // Check, but not accumulation, will happen even when firing normal rounds,
+    // so you can screw up your MP5 by firing blackpowder rounds and the
+    // effects will be in force even when firing retail ammo, though the gun
+    // won't get MORE foul.
+    if( ( it.dirt > 100 && one_in( ( ( 501 - it.dirt ) ) / 40 ) ) ||
+        ( it.faults.count(
+              fault_gun_clogged ) ) ) { // chance to cause malfunction, which always happens when clogged
+        if( !it.faults.count( fault_gun_clogged ) ) {
+            it.faults.insert( fault_gun_clogged );
+        }
+        if( it.faults.count( fault_gun_blackpowder ) ) {
+            it.faults.erase( fault_gun_blackpowder );
+        }
+        add_msg_player_or_npc( _( "Your foul %s misfires with a muffled click!" ),
+                               _( "<npcname>'s foul %s misfires with a muffled click!" ),
+                               it.tname() );
+        if( it.damage() < it.max_damage() && it.dirt > 350 &&
+            one_in( ( ( 505 - it.dirt ) ) / 30 ) ) { // chance to damage gun
+            add_msg_player_or_npc( m_bad, _( "Your %s is damaged by the blackpowder fouling accumulation!" ),
+                                   _( "<npcname>'s %s is damaged by the blackpowder fouling accumulation!" ),
+                                   it.tname() );
```

Comment:
Does this work correctly? I don't think string_format will work with this.

Reply:
I just pulled it and am trying to see. I think these have got to be added by either VS or astyle. It didn't happen back when I was working with notepad++/browser.

Reply:
I guess it works. No idea how to do a test with an NPC. Never done anything with NPCs except murdering them. 
![image](https://user-images.githubusercontent.com/45959491/62001666-5dc7d780-b0c3-11e9-9744-44e010d00e27.png)